### PR TITLE
Add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by naisofly**
Edited to match the current [CONTRIBUTING.md template](https://github.com/surrealdb/template/blob/main/CONTRIBUTING.md)

## What is the motivation?

Maintaining the same Contributing Guidelines across all SurrealDB repos

## What does this change do?

Updates the current guides to match the agreed-upon template for all SurrealDB repos

## Is this related to any issues?

It should give contributors raising issues a better idea of how we look at contributions from the community.
tl;dr: Discord Chat/Github Discussions -> GitHub Issue -> GitHub PR

## Have you read the [Contributing Guidelines](https://github.com/surrealdb/surrealdb.js/blob/main/CONTRIBUTING.md)?

- [x] I have read the [Contributing Guidelines](https://github.com/surrealdb/surrealdb.js/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+  
+  There isn't any specific configuration per driver. We may introduce configuration options in due course, and we will update this guide if we change those configurations.
+  
+  <!--
+  --------------------------------------------------
+  END
+  --------------------------------------------------
+  -->
+</details>
+
```
Comment (Ekwuno) [inline CONTRIBUTING.md:52]: This bit might not be relevant to contributing to the website repo.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+## Introducing new features
+
+We would &nbsp;<img width="15" alt="Love" src="https://github.com/surrealdb/surrealdb/blob/main/img/love.svg?raw=true">&nbsp; for you to contribute to SurrealDB, but we would also like to make sure SurrealDB is as great as possible and loyal to its vision and mission statement. For us to find the right balance, please open a question on [GitHub discussions](https://github.com/surrealdb/surrealdb/discussions) with any ideas before creating a [**GitHub Issue**](/issues). This will allow the SurrealDB community to have sufficient discussion about the new feature value and how it fits in the product roadmap and vision, before introducing a new pull request
```
Comment (Ekwuno) [inline CONTRIBUTING.md:55]: The link here is to the main SurrealDB discussion might be hard to track website specific features. Maybe we should use issues in this case.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+This is also important for the SurrealDB lead developers to be able to give technical input and different emphasis regarding the feature design and architecture. Some bigger features might need to go through our [RFC process](https://github.com/surrealdb/rfcs).
```
Comment (Ekwuno) [inline CONTRIBUTING.md:57]: Here as well

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+```
+bugfix-548-ensure-queries-execute-sequentially
+```
+
+Where `TYPE` can be one of the following:
+
+- **refactor** - code change that neither fixes a bug nor adds a feature
+- **feature** - code changes that add a new feature
+- **bugfix** - code changes that fix a bug
+- **docs** - documentation only changes
```
Comment (Ekwuno) [inline CONTRIBUTING.md:76]: We will want to redirect to https://github.com/surrealdb/docs.surrealdb.com/issues/new/choose

File: CONTRIBUTING.md (hunk 2 above)
Comment (naisofly) [reply CONTRIBUTING.md:55]: Hmm we want to encourage the pipeline of open discussion on Github/Discord -> GH Issue -> GH PR, rather than directly opening an issue.
Do you have a suggestion to rewrite here @Ekwuno ?
